src/lein_monolith/task/each.clj

```diff
@@ -86,7 +121,76 @@
           (if (get-in subproject [:profiles :monolith/inherited])
             [:default :monolith/inherited]
             [:default])))
-      (lein/apply-task (first task) subproject (rest task)))))
+      (config/debug-profile "apply-task"
+        (lein/resolve-and-apply subproject task)))))
+
+
+(defn- run-task!
+  "Runs the given task, returning a map of information about the run."
+  [ctx i target]
+  ; Try to reclaim some memory before running the task.
+  (System/gc)
+  (let [start (System/nanoTime)
+        opts (:opts ctx)
+        subproject (get-in ctx [:subprojects target])
+        results (delay {:name target
+                        :index i
+                        :elapsed (/ (- (System/nanoTime) start) 1000000.0)})]
+    (try
+      (lein/info (format "\nApplying to %s (%s/%s)"
```

Comment:
When running parallel tasks, this message prints `Applying to <project> (<n>/<N>)` where `n`, I assume, represents the number of that task in one possible topological ordering of the tasks, which doesn't reflect how the thread pool might decide to order the work. I recommend changing this behavior, and keeping a global counter that is incremented when a new task starts, so the number reflects the actual order in which tasks are started. This way it will still be obvious by looking at the output how much progress has been made.
```
Applying to project-A (1/79)
Applying to project-C (2/79)
Applying to project-B (3/79)
...
```
To go one step further, it would actually probably make the most sense to increment the counter and display the number *after* the task completes. That would better reflect how much progress has been made in an asynchronous context.
```
Applying to project-A
Applying to project-B
Applying to project-C
...
project-B complete (1/79)
...
project-C complete (2/79)
...
project-A complete (3/79)
```